### PR TITLE
Add `exclude` option

### LINKED ISSUE #54: Exclude some pokemon from random
**Is your feature request related to a problem? Please describe.**
Maybe #43 and #12

**Describe the solution you'd like**
I like this pokemon themes, they are very cute, I'm using with `pokemon: 'random'` and hyper-opacity plugin.
However, sometime it is hard to read the text because of its color.
So I want to exclude some pokemons from random, hou about to add `exclude` config option and exclude them?
If it is ok, I will make a PR.

**Additional context**
nothing

### PR DESCRIPTION
**Pull request by nasa9084**
Fixes #54

## Summary

This PR adds a new option `exclude`, which is for exclude some pokemon themes from random or category.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,6 +23,12 @@ function getUserOptions(configObj) {
       }
       return configObj.pokemon || 'pikachu';
     },
+    get exclude() {
+      if (Array.isArray(configObj.exclude)) {
+        return configObj.exclude;
+      }
+      return [configObj.exclude];
+    },
```
Comment (klaudiosinani) [inline index.js:38]:
```js
    get exclude() {
      const {exclude} = configObj;

      if (!exclude) {
        return [];
      }

      if (Array.isArray(exclude)) {
        return exclude;
      }

      return [exclude];
    },
```

File: index.js [hunk 2]
```diff
@@ -32,9 +38,12 @@ function getUserOptions(configObj) {
...
-  const index = Math.floor(Math.random() * (Object.keys(category).length));
-  const name = Object.keys(category)[index];
+function getRandomTheme(category, exclude) {
+  const keys = Object.keys(category).filter(key => {
+    return !exclude.includes(key);
+  });
+  const index = Math.floor(Math.random() * (keys.length));
+  const name = keys[index];
   return [name, category[name]];
 }
```
Comment (klaudiosinani) [inline index.js:59]:
```js
function getRandomTheme(themes, category, exclude) {
  const keys = Object.keys(category).filter(key => !exclude.includes(key));

  if (keys.length > 0) {
    const index = Math.floor(Math.random() * (keys.length));
    const name = keys[index];
    return [name, category[name]];
  }

  // ALl themes filtered out thus resolve to default
  return ['pikachu', themes.pokemon.pikachu];
}
```

File: index.js [hunk 3]
```diff
@@ -46,15 +55,15 @@ function getThemes() {
   return themes;
 }
 
-function getThemeColors(theme) {
+function getThemeColors(theme, exclude) {
   const themes = getThemes();
   const name = theme.trim().toLowerCase();
```
Comment (klaudiosinani) [inline index.js:71]:
```js
  const name = theme.trim().toLowerCase();
  const excludedThemes = [...new Set(exclude)].map(x => x.trim().toLowerCase());
```

File: index.js [hunk 4]
```diff
@@ -46,15 +55,15 @@ function getThemes() {
...
   if (name === 'random') {
-    return getRandomTheme(themes.pokemon);
+    return getRandomTheme(themes.pokemon, exclude);
   }
   if (Object.prototype.hasOwnProperty.call(themes, name)) {
     // Choose a random theme from the given category -- i.e. `fire`
-    return getRandomTheme(themes[name]);
+    return getRandomTheme(themes[name], exclude);
```
Comment (klaudiosinani) [inline index.js]:
```js
    return getRandomTheme(themes, themes[name], excludedThemes);
```

File: index.js [hunk 5]
```diff
@@ -46,15 +55,15 @@ function getThemes() {
   return themes;
 }
 
-function getThemeColors(theme) {
+function getThemeColors(theme, exclude) {
   const themes = getThemes();
   const name = theme.trim().toLowerCase();
   if (name === 'random') {
-    return getRandomTheme(themes.pokemon);
+    return getRandomTheme(themes.pokemon, exclude);
```
Comment (klaudiosinani) [inline index.js]:
```js
    return getRandomTheme(themes, themes.pokemon, excludedThemes);
```

**Review by klaudiosinani — CHANGES_REQUESTED** [review]
Great work, couple of tweaks and it is good to go
Also, the new `exclude` option must be added to the documentation; https://github.com/klaussinani/hyper-pokemon#options

**Comment by nasa9084** [comment]
@klaussinani Thank you for your comments, I applied them, PTAL

**Comment by klaudiosinani** [comment]
Looks great! Thank you a lot for taking the time to contribute!

**Pull request merged**